lib/puppet/type/file/source.rb

```diff
@@ -350,7 +350,12 @@ def chunk_file_from_source
         overwritten.
     EOT
 
-    defaultto :ignore
     newvalues(:use, :use_when_creating, :ignore)
+     munge do |value|
+      if @resource.file && @resource.line && !value.nil?
+        Puppet.puppet_deprecation_warning("The `source_permissions` parameter is deprecated. Explicitly set `owner`, `group`, and `mode`.", file: @resource.file, line: @resource.line)
```

Comment:
Should this string be localized? it is user facing.

Reply:
yep